corelib/frmts/crayfish_tifs.cpp

```diff
@@ -0,0 +1,116 @@
+ /*
+Crayfish - A collection of tools for TUFLOW and other hydraulic modelling packages
+Copyright (C) 2016 Lutra Consulting
+
+info at lutraconsulting dot co dot uk
+Lutra Consulting
+23 Chestnut Close
+Burgess Hill
+West Sussex
+RH15 8HN
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+
+#include "crayfish.h"
+#include "crayfish_mesh.h"
+#include "crayfish_dataset.h"
+#include "crayfish_output.h"
+#include "crayfish_gdal.h"
+
+#include <QString>
+#include <QtGlobal>
+#include <QFileInfo>
+#include <QStringList>
+#include <QDir>
+#include <QRegExp>
+#include <QDateTime>
+
+class MultipleTifsReader: public CrayfishGDALReader
+{
+public:
+    // We assume we have a directory with the multiple TIF files
+    // that have the same naming convention: filename_<timestamp>.asc
+    // each file is readable by GDAL reader and has one band with data
+    MultipleTifsReader(const QString& fileName): CrayfishGDALReader(fileName, "NETCDF"){}
+
+    QStringList parseDatasetNames(const QString& fileName) {
+        QStringList ret;
+
+        QFileInfo fi = QFileInfo(fileName);
+        QDir fd = fi.absoluteDir();
+
+        QStringList filters;
+        filters << "*.asc";
+
+        QStringList files = fd.entryList(filters, QDir::Files|QDir::NoSymLinks|QDir::NoDotAndDotDot|QDir::Readable, QDir::Name);
+        foreach (QString file, files) {
+            ret.push_back(fd.filePath(file));
+        }
+
+        return ret;
+    }
+
+    bool parseBandInfo(const CrayfishGDALDataset* cfGDALDataset,
+                       const metadata_hash& metadata, QString& band_name,
+                       float* time, bool* is_vector, bool* is_x
+                       ) {
+       static int err_counter = 0;
+
+       QFileInfo fi = QFileInfo(cfGDALDataset->mDatasetName);
+
+       // we except filename to be <prefix><YYYYMMDD_HHMM><suffix>.asc
+       // e.g. 5FQPE_20140819_0705_20140819020010_asc.asc
+       QRegExp rx("^(.*)(\\d{4})(\\d{2})(\\d{2})_(\\d{2})(\\d{2}).*.asc$");
```

Comment:
If we support other formats than .asc, this regex should be probably updated - same for "filters" variable above